controllers/v2/unavailability/api.py

```diff
@@ -1,31 +1,33 @@
-
 from flask import Blueprint, request, jsonify
 from flask_restful import reqparse, Resource, fields, marshal_with, Api, inputs
 
-from domain import session_scope
+from domain import session_scope, UserType
 from repository.volunteer_unavailability_v2 import *
-
+from services.jwk import requires_auth, is_user_or_has_role
+from controllers.v2.v2_blueprint import v2_api
 
 edit_parser = reqparse.RequestParser()
-edit_parser.add_argument("userId", type=int, required=True)
-edit_parser.add_argument("eventId", type=int, required=True)
 edit_parser.add_argument("title", type=str)
 edit_parser.add_argument("start", type=inputs.datetime_from_iso8601)
 edit_parser.add_argument("end", type=inputs.datetime_from_iso8601)
 edit_parser.add_argument("periodicity", type=int)
 
-class volunteer_unavailability(Resource):
-    def patch(self):
+
+class VolunteerUnavailabilityV2(Resource):
+
+    @requires_auth
+    @is_user_or_has_role(None, UserType.ROOT_ADMIN)
+    def put(self, user_id, event_id):
         args = edit_parser.parse_args()
         with session_scope() as session:
-            success = edit_event(session, **args)
+            success = edit_event(session, user_id, event_id, **args)
             if success is True:
                 return {"message": "Updated successfully"}, 200
             elif success is False:
                 return {"message": "Event not found"}, 404
             else:
-                return {"message": "Unexpected Error Occurred"}, 400
+                return {"message": "Unexpected Error Occurred"}, 406
+
 
-volunteer_unavailability_v2_bp = Blueprint('volunteer_unavailability-v2', __name__)
-api = Api(volunteer_unavailability_v2_bp, "/event")
-api.add_resource(volunteer_unavailability, "/unavailability")
+v2_api.add_resource(VolunteerUnavailabilityV2, '/v2/volunteers/',
+                    '/v2/volunteers/<user_id>/unavailability/<event_id>')
```

Comment:
This endpoint has two path urls, user_id and event_id. It is good practice to put non-sensitive IDs in the URL, as under REST the URL should refer to a specific resource (i.e. user user_id, and their unavailability event_id)